### PR TITLE
refactor: add generic Set implementation

### PR DESCRIPTION
**Pull request by knqyf263**
## What
- Add generic `Set[T]` interface and its implementation using `map[T]struct{}` under the hood
- Add convenient functions like `New()`, `Clone()`, and set operations (Union, Intersection, Difference)
- Add linter rule to suggest using this `set` package instead of raw `map[T]struct{}`

## Why
- There are many occurrences of using `map[T]struct{}` as sets in our codebase, which could benefit from a more readable and feature-rich implementation
- While several existing libraries provide the set implementations, I chose to implement it in our project because:
  1. The implementation is sufficiently simple and maintainable
  2. Avoiding additional external dependencies keeps our project more manageable
  3. We can customize the implementation to fit our specific needs
  4. It's easier to modify and extend the functionality as needed

## Implementation Details
- The Set interface is generic and accepts any comparable type
- The implementation (`unsafeSet`) is not thread-safe by design, as that matches our current usage patterns

## Additional Information
- Added linter rule will help team members discover this package when they try to use `map[T]struct{}`
- The implementation is intentionally kept minimal but extensible
- Future improvements could include:
  - Thread-safe implementation if needed
  - Additional set operations (SymmetricDifference, IsSubset, etc.)

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/fanal/analyzer/imgconf/apk/apk_test.go [hunk 1]
```diff
@@ -1508,86 +1509,41 @@ func TestResolveDependency(t *testing.T) {
...
 		"package alias": {
 			pkgName:             "sqlite-dev",
 			apkIndexArchivePath: "testdata/history_v3.9.json",
-			expected: map[string]struct{}{
-				"sqlite-dev":  {},
-				"sqlite-libs": {},
-				"pkgconf":     {}, // pkgconfig => pkgconf
-				"musl":        {},
-			},
+			want: set.New(
```
Comment (DmitriyLewen) [inline pkg/fanal/analyzer/imgconf/apk/apk_test.go:1535]: nit: Let's use same fmt.

**Review by DmitriyLewen — APPROVED** [review]
LGTM

File: pkg/fanal/analyzer/imgconf/apk/apk_test.go (hunk 1 above)
Comment (knqyf263) [reply pkg/fanal/analyzer/imgconf/apk/apk_test.go:1566]: We need to keep a comment in this test case. Instead, I reverted others. https://github.com/aquasecurity/trivy/pull/8149/commits/a69fd6645ff8bc2936e024426e454a3b4f83c235

File: pkg/report/table/vulnerability.go [hunk 2]
```diff
@@ -291,33 +292,32 @@ func (r *vulnerabilityRenderer) printf(format string, args ...any) {
...
 			}
 		}
 	}
 
 	// Omitted
-	rootIDs := lo.Filter(lo.Keys(roots), func(pkgID string, _ int) bool {
-		_, ok := seen[pkgID]
-		return !ok
+	rootIDs := lo.Filter(roots.Items(), func(pkgID string, _ int) bool {
+		return !seen.Contains(pkgID)
```
[the remark this answers is not in the document]
Comment (knqyf263) [reply pkg/report/table/vulnerability.go]: Nice catch. Fixed in https://github.com/aquasecurity/trivy/pull/8149/commits/7c04b80a28e9745f2430b957b263f2fcf902a1b5

**Review by nikpivkin — APPROVED** [review]

**Pull request merged**